File: benchmarks/6_mandelbrot/bench.c
/* Benchmark 6: Mandelbrot — 800x800 grid, 1000 max iterations */
#include <stdio.h>
#include <time.h>

int main(void) {
    clock_t t0 = clock();

    long long count = 0;
    for (int py = 0; py < 800; py++) {
        double cy = (py / 800.0) * 2.0 - 1.0;
        for (int px = 0; px < 800; px++) {
            double cx = (px / 800.0) * 3.5 - 2.5;
            double x = 0.0, y = 0.0;
            int iter = 0;
            while (iter < 1000 && x*x + y*y < 4.0) {
                double tx = x*x - y*y + cx;
                y = 2.0*x*y + cy;
                x = tx;
                iter++;
            }
            if (iter == 1000) count++;
        }
    }

    clock_t t1 = clock();
    long long ms = (long long)((double)(t1 - t0) / CLOCKS_PER_SEC * 1000.0);
    printf("%lld\n", count);
    printf("TIME_MS:%lld\n", ms);
    return 0;
}

File: tauraroc-Linux-X64/benchmarks/6_mandelbrot/bench.rs
// Benchmark 6: Mandelbrot — 800x800 grid, 1000 max iterations
use std::time::Instant;

fn main() {
    let t0 = Instant::now();

    let mut count: i64 = 0;
    for py in 0..800_i64 {
        let cy = (py as f64 / 800.0) * 2.0 - 1.0;
        for px in 0..800_i64 {
            let cx = (px as f64 / 800.0) * 3.5 - 2.5;
            let (mut x, mut y) = (0.0_f64, 0.0_f64);
            let mut iter = 0;
            while iter < 1000 && x * x + y * y < 4.0 {
                let tx = x * x - y * y + cx;
                y = 2.0 * x * y + cy;
                x = tx;
                iter += 1;
            }
            if iter == 1000 { count += 1; }
        }
    }

    let ms = t0.elapsed().as_millis();
    println!("{}", count);
    println!("TIME_MS:{}", ms);
}
